processor/marshalprocessor/processor.go

```diff
@@ -0,0 +1,110 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marshalprocessor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type marshalProcessor struct {
+	logger    *zap.Logger
+	marshalTo string
+}
+
+func newMarshalProcessor(logger *zap.Logger, cfg *Config) *marshalProcessor {
+	return &marshalProcessor{
+		logger:    logger,
+		marshalTo: cfg.MarshalTo,
+	}
+}
+
+// ErrStringBodyNotSupported is returned when a log body is a string
+var ErrStringBodyNotSupported = fmt.Errorf("String body not supported")
+
+func (mp *marshalProcessor) processLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	var errors []error
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			scopeLog := resourceLog.ScopeLogs().At(j)
+			for k := 0; k < scopeLog.LogRecords().Len(); k++ {
+				logRecord := scopeLog.LogRecords().At(k)
+				logBody := logRecord.Body()
+				// If body is a string, skip that log
+				if logBody.Type() == pcommon.ValueTypeStr {
+					errors = append(errors, ErrStringBodyNotSupported)
+					continue
+				}
+				switch strings.ToUpper(mp.marshalTo) {
+				case "JSON":
+					jsonBody := logBody.AsString()
+					logBody.SetStr(jsonBody)
+				case "XML":
+					return ld, fmt.Errorf("XML not yet supported")
+				case "KV":
+					jsonBody := logBody.AsString()
+					kvBody, err := convertJSONToKV(jsonBody)
+					if err != nil {
+						errors = append(errors, fmt.Errorf("Error converting to KV: %w", err))
+						continue
+					}
+					logBody.SetStr(kvBody)
+				default:
+					return ld, fmt.Errorf("Unrecognized format to marshal to: %s", mp.marshalTo)
+				}
+			}
+		}
+	}
+
+	if len(errors) == 0 {
+		return ld, nil
+	}
+	var errorStrings []string
+	for _, err := range errors {
+		errorStrings = append(errorStrings, err.Error())
+	}
+	return ld, fmt.Errorf("%s", strings.Join(errorStrings, ", "))
+}
+
+func convertJSONToKV(jsonBody string) (string, error) {
+	var data map[string]interface{}
+	err := json.Unmarshal([]byte(jsonBody), &data)
+	if err != nil {
+		return jsonBody, fmt.Errorf("Error unmarshalling JSON: %w", err)
+	}
+
+	var keyValuePairs []string
+	for key, value := range data {
+		strValue := fmt.Sprintf("%v", value)
+		keyValuePairs = append(keyValuePairs, fmt.Sprintf("%s=%s", key, strValue))
+	}
+
+	// Ensure consistent order for testing
+	sort.Slice(keyValuePairs, func(i, j int) bool {
+		return keyValuePairs[i] < keyValuePairs[j]
+	})
+
+	kvBody := strings.Join(keyValuePairs, " ")
```

Comment:
What happens if the key or value has a space in them already?

Reply:
Great question... not sure if the format has a standard way to represent spaces in the key or value. I will look into it.

Reply:
I think it will be an arbitrary escape/replacement, the only example of KV I have does not cover this case.

Reply:
I have added support for configurable separators and quoting keys/vals that contain those separators. 
Also improved support for marshaling KV from nested fields at Dylan's suggestion.